code/modules/detective_work/evidence.dm

```diff
@@ -80,3 +80,181 @@
+	if(supplied)
+		copy_evidence(supplied)
+		name = "[initial(name)] (\the [supplied])"
+
+/obj/item/sample/print/New(newloc, atom/supplied)
+	. = ..()
+	if(length(evidence))
+		icon_state = "fingerprint1"
+
+/obj/item/sample/proc/copy_evidence(atom/supplied)
+	if(length(supplied.suit_fibers))
+		evidence = supplied.suit_fibers.Copy()
+		supplied.suit_fibers.Cut()
+
+/obj/item/sample/proc/merge_evidence(obj/item/sample/supplied, mob/user)
+	if(!length(supplied.evidence))
+		return FALSE
+	evidence |= supplied.evidence
+	name = "[initial(name)] (combined)"
+	to_chat(user, "<span class='notice'>You are moving [supplied] to [src].</span>")
+	return TRUE
+
+/obj/item/sample/print/merge_evidence(obj/item/sample/supplied, mob/user)
+	if(!length(supplied.evidence))
+		return FALSE
+	for(var/print in supplied.evidence)
+		if(evidence[print])
+			evidence[print] = stringmerge(evidence[print], supplied.evidence[print])
+		else
+			evidence[print] = supplied.evidence[print]
+	name = "[initial(name)] (combined)"
+	to_chat(user, "<span class='notice'>You overlay [src] and [supplied], combining the print records.</span>")
+	return TRUE
+
+/obj/item/sample/pre_attack(atom/A, mob/living/user, params)
+	..()
+	// Fingerprints will be handled in after_attack() to not mess up the samples taken
+	return A.attackby__legacy__attackchain(src, user, params)
+
+/obj/item/sample/attackby__legacy__attackchain(obj/O, mob/user)
+	if(istype(O, src.type))
+		user.unequip(O)
+		if(merge_evidence(O, user))
+			qdel(O)
+		return TRUE
+	return ..()
+
+/obj/item/sample/fibers
+	name = "fiber bag"
+	desc = "Used to store fiber evidence for forensic examianation."
+	icon_state = "fiberbag"
+
+/obj/item/sample/print
+	name = "fingerprint card"
+	desc = "Preserves fingerprints."
+	icon = 'icons/obj/card.dmi'
+	icon_state = "fingerprint0"
+	item_state = "paper"
+	var/used
+
+/obj/item/sample/print/update_icon_state()
+	. = ..()
+	icon_state = "fingerprint0"
+	if(used)
+		icon_state = "fingerprint1"
+
+
+/obj/item/sample/print/attack_self__legacy__attackchain(mob/user)
+	if(!length(evidence))
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.gloves)
+		to_chat(user, "<span class='warning'>Take [H.gloves] off first.</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You press your fingertips firmly against the card.</span>")
+	var/fullprint = H.get_full_print()
+	evidence[fullprint] = fullprint
+	name = "[initial(name)] ([H])"
+	used = TRUE
+	update_appearance(UPDATE_ICON_STATE)
+
+/obj/item/sample/print/attack__legacy__attackchain(mob/living/M, mob/living/user, def_zone)
+	. = ..()
+
+	if(!ishuman(M))
+		return ..()
+
+	if(length(evidence))
+		return FALSE
+
+	var/mob/living/carbon/human/H = M
+
+	if(H.gloves)
+		to_chat(user, "<span class='warning'>[H] is wearing gloves.</span>")
+		return TRUE
+
+	if(user != H && !IS_HORIZONTAL(H))
+		user.visible_message("<span class='danger'>[user] tried to fingerprint [H], but he resists.</span>")
```

Comment:
I think forcing them to be horizontal is a little odd? I think it might be better if you very clearly give them the chance to resist if they aren't horizontal, since otherwise you'll need to be like "lay down sir so I can fingerprint you."

What I would do here would be to do something like an injection: Giving a target a very loud telegraph that fingerprinting is happening, and let them walk away and reject it, but let them take it by standing in place if they want.

Something like:
`target.visible_message("<span class='notice'>[user] tries to take [H]'s fingerprints.</span>", "<span class='userdanger'>[user] forces your fingers onto [src], trying to take your fingerprints!</span>")`, with another message in the failed do_after if either of them move.